app/lib/features/onboarding/pages/redeem_invitations_page.dart

```diff
@@ -0,0 +1,355 @@
+    final errorStyle = Theme.of(
+      context,
+    ).textTheme.labelMedium?.copyWith(color: errorColor);
+    final lang = L10n.of(context);
+    final token = _tokenController.text.trim();
+
+    if (token.isEmpty) {
+      return const SizedBox.shrink();
+    }
+
+    return ref
+        .watch(superInviteInfoProvider(token))
+        .when(
+          data: (info) {
+            EasyLoading.dismiss();
+            setState(() {
+              showRedeemButton = true;
+              showGetDetailsButton = false;
+            });
+            return _renderInfo(context, info);
+          },
+          error: (e, s) {
+            EasyLoading.dismiss();
+            setState(() {
+              showRedeemButton = false;
+              showGetDetailsButton = true;
+            });
+            final errorStr = e.toString();
+            if (errorStr.contains('error: [404]')) {
+              return Text(
+                lang.superInvitesPreviewMissing(token),
+                style: errorStyle,
+              );
+            }
+            if (errorStr.contains('error: [403]')) {
+              return Text(lang.superInvitesDeleted(token), style: errorStyle);
+            }
+            return Text(lang.loadingFailed(e), style: errorStyle);
+          },
+          loading: () {
+            setState(() {
+              showRedeemButton = false;
+              showGetDetailsButton = true;
+            });
+            EasyLoading.show(status: lang.loading);
+            return const SizedBox.shrink();
+          },
+        );
+  }
+
+  Widget _renderInfo(BuildContext context, SuperInviteInfo info) {
+    final lang = L10n.of(context);
+    final displayName = info.inviterDisplayNameStr();
+    final userId = info.inviterUserIdStr();
+    final inviter = displayName ?? userId;
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Card(
+          margin: EdgeInsets.zero,
+          child: ListTile(
+            title: Text(inviter),
+            subtitle: Text(lang.youInvited(info.roomsCount())),
+            leading: ActerAvatar(
+              options: AvatarOptions.DM(
+                AvatarInfo(uniqueId: userId, displayName: displayName),
+                size: 18,
+              ),
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildNavigationButtons(BuildContext context, L10n lang) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        showGetDetailsButton
+            ? _buildGetDetailsButton(context, lang)
+            : const SizedBox.shrink(),
+        const SizedBox(height: 16),
+        showRedeemButton
+            ? Column(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: [
+                _buildRedeemButton(context, lang),
+                const SizedBox(height: 16),
+              ],
+            )
+            : const SizedBox.shrink(),
```

Comment:
Why to use `Column` widget again if we are already in `Column` hierarchy. ?

```suggestion
        if(showRedeemButton) ...[
               _buildRedeemButton(context, lang),
               const SizedBox(height: 16),
         ],
```

Reply:
ok, sure will do that change.